### DISCUSSION TITLE
What are environment variables, and how to define them?

### DISCUSSION THREAD

**Discussion by JRBros2346** [slint-ui/slint#3300] [Q&A]
I'm using rust in windows. I'm getting this error, when I build my app
environment variable `SLINT_INCLUDE_GENERATED` not defined at compile time
What does it mean, and how to define them at compile time?
Also what are all the environment variables? Are they like C++style macros?
I found `SLINT_STYLE` too, but don't know how to use it.

**Comment by ogoffart** [comment]
This env variable is used to communicate between slint_build and the  include_modules! macro.
In this case, it probably means that build.rs didn't generate the rust code from your .slint files.

Your build.rs should be looking something like this:

```rust
fn main() {
    slint_build::compile("ui/main.slint").unwrap();
}
```

**Comment by ogoffart** [comment]
To anwser the more generic question about what are environment variables, the Wikipedia page has some good information: https://en.wikipedia.org/wiki/Environment_variable#Assignment:_DOS,_OS/2_and_Windows
